gm4_vecto_shamir/data/gm4_vecto_shamir/predicates/check_water/below.json

```diff
@@ -0,0 +1,64 @@
+[  
+  {
+    "condition": "minecraft:alternative",
+    "terms": [
+      {
+        "condition": "minecraft:location_check",
+        "offsetX": 0,
+        "offsetY": -1,
+        "offsetZ": 0,
+        "predicate": {
+          "block": {
+            "tag": "gm4_vecto_shamir:boat_spawnable"
+          }
+        }
+      },
+      {
+        "condition": "minecraft:location_check",
+        "offsetX": 0,
+        "offsetY": -1,
+        "offsetZ": 0,
+        "predicate": {
+          "block": {
+            "tag": "minecraft:slabs",
+            "state": {
+              "type": "bottom",
+              "waterlogged": "true"
+            }
+          }
+        }
+      },
+      {
+        "condition": "minecraft:location_check",
+        "offsetX": 0,
+        "offsetY": -1,
+        "offsetZ": 0,
+        "predicate": {
+          "block": {
+            "tag": "minecraft:fence_gates",
+            "state": {
+              "open": "true",
+              "waterlogged": "true"
+            }
+          }
+        }
+      },
+      {
+        "condition": "minecraft:location_check",
+        "offsetX": 0,
+        "offsetY": -1,
+        "offsetZ": 0,
+        "predicate": {
+          "block": {
+            "tag": "minecraft:trapdoors",
+            "state": {
+              "open": "false",
+              "half": "bottom",
+              "waterlogged": "true"
```

Comment:
Instead of listing a couple of waterloggable blocks, these should probably use `#gm4:waterloggable` to make the module more maintainable.
The same goes for the other predicates in `check_water/`.

Reply:
Most blocks in #gm4:waterloggable are not valid spawning spaces for boats. The blocks it checks for shouldn't intersect with the spawning area of the boat (bottom slabs and trapdoors are fine for example, but their "top" versions are not)

Reply:
That is a valid point, though we are trading accuracy for maintainability.